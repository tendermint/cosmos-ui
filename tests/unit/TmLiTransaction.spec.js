import { mount } from "@vue/test-utils"
import { TmLiTransaction } from "@/index"

describe("TmLiTransaction", () => {
  let wrapper
  let propsData = {
    color: "#FFFFFF",
    time: 1003442,
    block: 500
  }

  beforeEach(() => {
    wrapper = mount(TmLiTransaction, {
      propsData,
      slots: {
        caption: "<span>Some Caption</span>",
        details: "<span>Some Details</span>"
      }
    })
  })

  it("has the expected html structure", () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
