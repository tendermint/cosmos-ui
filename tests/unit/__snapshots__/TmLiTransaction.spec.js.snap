// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmLiTransaction has the expected html structure 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #ffffff;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <p
      class="tm-li-tx__content__caption"
    >
      <span>
        Some Caption
      </span>
    </p>
    <div
      class="tm-li-tx__content__information"
    >
      <div
        class="tm-li-tx__content__information__details"
      >
        <span>
          Some Details
        </span>
      </div>
      <div
        class="tm-li-tx__content__information__block"
      >
        <a
          href="https://explorecosmos.network/blocks/500"
        >
          Block #500Â 
        </a>
        @ 2:03 am
      </div>
    </div>
  </div>
</div>
`;
